src/lib/datepicker/calendar.ts

```diff
@@ -44,7 +44,18 @@ import {MatMonthView} from './month-view';
 import {MatMultiYearView, yearsPerPage, yearsPerRow} from './multi-year-view';
 import {MatYearView} from './year-view';
 import {Directionality} from '@angular/cdk/bidi';
+import {ComponentPortal, ComponentType, Portal} from '@angular/cdk/portal';
 
+/** Default header for MatCalendar */
+@Component({
```

Comment:
lint: `Missing required properties: encapsulation, moduleId, preserveWhitespaces, changeDetection` (can set them all to the same things `MatCalendar` below)

Reply:
done in https://github.com/angular/material2/pull/9639/commits/a45f3ee2748ef52aa494ee7938bc5a0f43a5a18a